### ISSUE TITLE
MODEL_ZOO: as Baseline  for training on a self created dataset 

### ISSUE THREAD

**Issue by Idolized22** [facebookresearch/maskrcnn-benchmark#541]
## ❓ Questions and Help

Hello 
How can I use weights from  model zoo as input weights on a self created dataset ? 

what changes should be made in order to do so? 

I am interested in using the X-101-32x8d-FPN mask model. 

Thanks for helping in advance !

**Comment by chengyangfu** [comment]
Hi, 
Please check this section 
https://github.com/facebookresearch/maskrcnn-benchmark#finetuning-from-detectron-weights-on-custom-datasets  and the referenced issue, it has a lot of discussion on how to approach this problem.

**Comment by fmassa** [comment]
Closing this from @chengyangfu comment

**Comment by Idolized22** [comment]
@chengyangfu 
> Hi,
> Please check this section
> https://github.com/facebookresearch/maskrcnn-benchmark#finetuning-from-detectron-weights-on-custom-datasets and the referenced issue, it has a lot of discussion on how to approach this problem.

Thank you for your quick replay, 

I have checked the mentioned discussion before opening this issue , 
but since I am interested in a model from the model zoo of type pth I am not sure if any modification are needed. 
especially regarding the last layers.

**Comment by chengyangfu** [comment]
Hi, 
   You still need to modify the model.  
    There are two ways to handle this problem. First is just removing the last layer. If some of the classes in your dataset are overlapped with the COCO classes, the better way is to do the net surgery to keep specific weights in the last layer. 

   For the first one, once you downloaded the model from [Model Zoo](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/MODEL_ZOO.md), you can use methods in this thread https://github.com/facebookresearch/maskrcnn-benchmark/issues/430 to remove the last layer. 
    If you decide to do some net surgery, you could check the [code](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/tools/cityscapes/convert_cityscapes_to_coco.py). P.S. It reads the Caffe2 model instead of PyTorch model.  You need to change parts of the code to make it work.
